subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm_test.go

```diff
@@ -484,6 +485,27 @@ func testDeleteAllocatedPods(t *testing.T, h *HivedAlgorithm) {
 	}
 }
 
+func testSuggestedNodes(t *testing.T, h *HivedAlgorithm) {
+	var idx int32
+	for i, node := range allNodes {
```

Comment:
>allNodes [](start = 22, length = 8)

Just clone to suggestednodes slice with the node removed, instead of inplace change allnodes.

Tests may be run concurrently.